pkg/image/api/types.go

```diff
@@ -157,3 +205,75 @@ type DockerImageReference struct {
 	Tag       string
 	ID        string
 }
+
+// ImageStreamImport allows a caller to request information about a set of images for possible
+// import into an image stream, or actually tag the images into the image stream.
+type ImageStreamImport struct {
+	unversioned.TypeMeta
+	// ObjectMeta must identify the name of the image stream to create or update. If resourceVersion
+	// is set, the object will be assumed to alreday exist.
+	kapi.ObjectMeta
+
+	// Spec is the set of items desired to be imported
+	Spec ImageStreamImportSpec
+	// Status is the result of the import
+	Status ImageStreamImportStatus
+}
+
+// ImageStreamImportSpec defines what images should be imported.
+type ImageStreamImportSpec struct {
+	// Import indicates whether to perform an import - if so, the specified tags are set on the spec
+	// and status of the image stream defined by the type meta.
+	Import bool
+	// Repository is an optional import of an entire Docker image repository. A maximum limit on the
+	// number of tags imported this way is imposed by the server.
+	Repository *RepositoryImportSpec
+	// Images are a list of individual images to import.
+	Images []ImageImportSpec
+}
+
+// ImageStreamImportStatus contains information about the status of an image stream import.
+type ImageStreamImportStatus struct {
+	// Import is the image stream that was successfully updated or created when 'to' was set.
```

Comment:
But I'm going to skip it for now, because it's a larger refactor of the behavior.  I'll open a follow up.